docs/notebooks/.ipynb_checkpoints/WMD_tutorial-checkpoint.ipynb

```diff
@@ -0,0 +1,601 @@
+    "\n",
+    "WMD uses word2vec vector embeddings [2] of words (read about word2vec in Gensim [here](http://rare-technologies.com/deep-learning-with-word2vec-and-gensim/) and [here](http://rare-technologies.com/word2vec-tutorial/)).\n",
+    "\n",
+    "WMD is illustrated below for two very similar sentences. The sentences have no words in common, but by matching the relevant words, WMD is able to accurately measure the similarity between the two sentences. The method also uses the bag-of-words representation of the documents (simply put, the word's frequencies in the documents), noted as $d$ in the figure below. The intution behind the method is that we find the minimum \"traveling distance\" between documents, the most efficient way to \"move\" the distribution of document 1 to the distribution of document 2.\n",
+    "\n",
+    "<img src='https://vene.ro/images/wmd-obama.png' height='600' width='600'>\n",
+    "\n",
+    "\n",
+    "This method was introduced in the article \"From Word Embeddings To Document Distances\" by Matt Kusner et al. ([link to PDF](http://jmlr.org/proceedings/papers/v37/kusnerb15.pdf)). It is inspired by the \"Earth Mover's Distance\", and employs a solver of the \"transportation problem\".\n",
+    "\n",
+    "In this tutorial, we will learn how to use Gensim's WMD functionality, which consists of the `wmdistance` method for distance computation, and the `WmdSimilarity` class for corpus based similarity queries.\n",
+    "\n",
+    "> **Note**:\n",
+    ">\n",
+    "> If you use this software, please consider citing the following papers:\n",
+    ">\n",
+    "> Ofir Pele and Michael Werman, \"A linear time histogram metric for improved SIFT matching\".\n",
+    "> \n",
+    "> Ofir Pele and Michael Werman, \"Fast and robust earth mover's distances\".\n",
+    ">\n",
+    "> Matt Kusner et al. \"From Word Embeddings To Document Distances\"."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> **Running this notebook:**\n",
+    ">\n",
+    "> You can download this [iPython Notebook](http://ipython.org/notebook.html), and run it on your own computer, provided you have installed Gensim and NLTK, and downloaded the necessary data.\n",
+    ">\n",
+    "> The notebook was run on an Ubuntu machine with an Intel core i7-4770 CPU 3.40GHz (8 cores) and 32 GB memory. Running the entire notebook on this machine takes about 5 minutes.\n",
+    "\n",
+    "## Using WMD\n",
+    "\n",
+    "To use WMD, we need some word embeddings first of all. You could train a word2vec (see tutorial [here](http://rare-technologies.com/word2vec-tutorial/)) model on some corpus, but we will start by download some pre-trained word2vec embeddings. Download these embeddings [here](https://drive.google.com/file/d/0B7XkCwpI5KDYNlNUTTlSS21pQmM/edit). Training your own embeddings can be beneficial, but to simplify this tutorial, we will be using pre-trained embeddings at first.\n",
```

Comment:
Less fragile link is "download GoogleNews-vectors-negative300.bin.gz from https://code.google.com/archive/p/word2vec/ Warning: 1.5Gb"